static/app/views/automations/components/ruleNodeList.tsx

```diff
@@ -0,0 +1,58 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import {Select} from 'sentry/components/core/select';
+import type {DataCondition} from 'sentry/types/workflowEngine/dataConditions';
+import {dataConditionNodesMap} from 'sentry/views/automations/components/dataConditionNodes';
+import RuleNode from 'sentry/views/automations/components/ruleNode';
+
+interface RuleNodeListProps {
+  conditions: Array<Omit<DataCondition, 'condition_group' | 'type' | 'id'>>;
+  group: string;
+  onAddRow: (type: string) => void;
+  onDeleteRow: (id: number) => void;
+  placeholder: string;
+  updateCondition: (index: number, condition: Record<string, any>) => void;
+}
+
+export default function RuleNodeList({
+  group,
+  placeholder,
+  conditions,
+  onAddRow,
+  onDeleteRow,
+  updateCondition,
+}: RuleNodeListProps) {
+  const options = Object.entries(dataConditionNodesMap).map(([value, node]) => ({
+    value,
+    label: node.label,
+  }));
+
+  return (
+    <Fragment>
+      {Object.entries(conditions).map(([i, condition]) => (
```

Comment:
I think you can just use `Array.map` here and avoid `parseInt(i, 10)`?
```suggestion
      {conditions.map((condition, i) => (
```